air/src/constraints/range.rs

```diff
@@ -113,48 +118,39 @@ pub fn enforce_aux_constraints<F, E>(
     let alpha = aux_rand_elements.get_segment_elements(0)[0];
 
     // Enforce b_range.
-    enforce_running_product_b_range(main_frame, aux_frame, alpha, &mut result[..]);
+    enforce_b_range(main_frame, aux_frame, alpha, result);
 }
 
 // TRANSITION CONSTRAINT HELPERS
 // ================================================================================================
 
-// --- MAIN TRACE ---------------------------------------------------------------------------------
-
-/// Constrain the selector flags to binary values.
-fn enforce_flags<E: FieldElement>(frame: &EvaluationFrame<E>, result: &mut [E]) -> usize {
-    let constraint_count = 2;
-
-    result[0] = is_binary(frame.s0());
-    result[1] = is_binary(frame.s1());
-
-    constraint_count
-}
-
-/// Constrain the transition between rows in the range checker table.
-fn enforce_delta<E: FieldElement>(frame: &EvaluationFrame<E>, result: &mut [E]) -> usize {
-    let constraint_count = 1;
-
-    result[0] = frame.change(V_COL_IDX)
-        * (frame.change(V_COL_IDX) - E::ONE)
-        * (frame.change(V_COL_IDX) - E::from(3_u8))
-        * (frame.change(V_COL_IDX) - E::from(9_u8))
-        * (frame.change(V_COL_IDX) - E::from(27_u8))
-        * (frame.change(V_COL_IDX) - E::from(81_u8))
-        * (frame.change(V_COL_IDX) - E::from(243_u8))
-        * (frame.change(V_COL_IDX) - E::from(729_u16))
-        * (frame.change(V_COL_IDX) - E::from(2187_u16));
-
-    constraint_count
-}
-
 // --- AUXILIARY COLUMNS (FOR MULTISET CHECKS) ----------------------------------------------------
 
-/// Ensures that the running product is computed correctly in the column `b_range`. It enforces
-/// that the value only changes after the padded rows, where the value of `z` is included at each
-/// step, ensuring that the values in the range checker table are multiplied into `b_range` 0, 1,
-/// 2, or 4 times, according to the selector flags.
-fn enforce_running_product_b_range<E, F>(
+/// Ensures that the range checker bus is computed correctly. It enforces an implementation of the
+/// LogUp lookup as a running sum "bus" column. All values in the range checker trace are saved
+/// with their lookup multiplicity and the logarithmic derivatives are added to b_range. Values
+/// for which lookups are requested from the stack and memory are each looked up with multiplicity
+/// one, and the logarithmic derivatives are subtracted from b_range.
+///
+/// Let the following variables:
+/// - rc_value: the range checker value
+/// - rc_multiplicity: the range checker multiplicity value
+/// - flag_s: boolean flag indicating a stack operation with range checks
+/// - sv0-sv3: stack value 0-3, the 4 values range-checked from the stack
+/// - flag_m: boolean flag indicating the memory chiplet is active (i.e. range checks are required)
+/// - mv0-mv1: memory value 0-1, the 2 values range-checked from the memory chiplet
```

Comment:
nit: one thing that might be helpful to note here is the degrees of `flag_s` and `flag_m`. I believe these are 3 for both - but I had look in a couple of places to figure it out.